vllm/model_executor/models/__init__.py

```diff
@@ -189,12 +190,21 @@ def get_supported_archs() -> List[str]:
         return list(_MODELS.keys()) + list(_OOT_MODELS.keys())
 
     @staticmethod
-    def register_model(model_arch: str, model_cls: Type[nn.Module]):
+    def register_model(model_arch: str,
+                       model_cls: Type[nn.Module],
+                       is_multimodal: bool = False):
         if model_arch in _MODELS:
             logger.warning(
                 "Model architecture %s is already registered, and will be "
                 "overwritten by the new model class %s.", model_arch,
                 model_cls.__name__)
+
+        # NOTE: This is needed to store the information if the OOT model is
+        # an multimodal model.
+        if is_multimodal:
+            global _OOT_MULTIMODAL_MODELS
+            _OOT_MULTIMODAL_MODELS[model_arch] = model_cls
+
         global _OOT_MODELS
```

Comment:
do you need an `else` branch here?

Reply:
No, we use `_OOT_MULTIMODAL_MODELS` to only check if the model is multimodal, and we still use `_OOT_MODELS` to check if the model is actually supported.

This follows the same relationship between `_MODELS` and `_MULTIMODAL_MODELS`.